core/tools/blueos_startup_update/blueos_startup_update.py

```diff
@@ -438,6 +438,79 @@ def ensure_user_data_structure_is_in_place() -> bool:
     return False
 
 
+def load_pi5_overlays_in_runtime() -> bool:
+    loaded_overlays = run_command("dtoverlay -l", check=False).stdout.split("\n")
+    loaded_overlays = [overlay.split(": ")[1].strip() for overlay in loaded_overlays if ": " in overlay]
+    loaded_overlay_names = [overlay.split(" ")[0] for overlay in loaded_overlays if "dtparam" not in overlay]
+    loaded_dt_params = [overlay.split(" ")[-1] for overlay in loaded_overlays if "dtparam" in overlay]
```

Comment:
I would recommend using a simple regex over doing this logic, it'll be simpler and more readable.

`^(?<index>\d):\s+(?<name>\S+)($|\s+(?<args>.*))`

Reply:
I agree with Patrick here, it's easier to follow the parsing from the regex line.